src/ast.ts

```diff
@@ -36,20 +36,26 @@ export interface ElementNodeProto {
   children: Array<NodeProto>;
 }
 
+export interface DocumentNodeProto {
+  tagid: 92; // See htmltagenum.ts
+  children: Array<NodeProto>;
+}
+
 export type NodeProto = TextNodeProto | ElementNodeProto;
 export interface AttributeProto {
   name: string;
   value?: string;
 }
 
 export function isElementNode(node: NodeProto): node is ElementNodeProto {
-  return 'tagid' in node;
+  return (node as any).tagid !== undefined;
```

Comment:
Why is the `any` needed here?

Reply:
Because `tagid` only exists `ElementNodeProto`, not on `NodeProto` objects